MediaBrowser.Providers/Music/AlbumMetadataService.cs

```diff
@@ -72,15 +74,32 @@ private ItemUpdateType SetAlbumArtistFromSongs(MusicAlbum item, IEnumerable<Audi
         {
             var updateType = ItemUpdateType.None;
 
-            var artists = songs
+            var albumArtists = songs
                 .SelectMany(i => i.AlbumArtists)
-                .Distinct(StringComparer.OrdinalIgnoreCase)
-                .OrderBy(i => i)
+                .GroupBy(i => i)
+                .OrderByDescending(g => g.Count())
+                .Select(g => g.Key)
+                .ToArray();
+
+            var musicbrainzAlbumArtistIds = songs
+                .Select(i => i.GetProviderId(MetadataProvider.MusicBrainzAlbumArtist))
+                .GroupBy(i => i)
+                .OrderByDescending(g => g.Count())
+                .Select(g => g.Key)
                 .ToArray();
 
-            if (!item.AlbumArtists.SequenceEqual(artists, StringComparer.OrdinalIgnoreCase))
+            var musicbrainzAlbumArtistId = item.GetProviderId(MetadataProvider.MusicBrainzAlbumArtist);
+            if (!String.IsNullOrEmpty(musicbrainzAlbumArtistIds[0]) &&
+                (String.IsNullOrEmpty(musicbrainzAlbumArtistId)
+                    || !musicbrainzAlbumArtistId.Equals(musicbrainzAlbumArtistIds[0], StringComparison.OrdinalIgnoreCase)))
+            {
+                item.SetProviderId(MetadataProvider.MusicBrainzAlbumArtist, musicbrainzAlbumArtistIds[0]!);
```

Comment:
If you pull `musicbrainzAlbumArtistIds[0]` out to a variable the null-forgiving operator isn't needed